### DISCUSSION TITLE
lance has no module 'dataset'

### DISCUSSION THREAD

**Discussion by Aryan-Deshpande** [lance-format/lance#2050] [Q&A]
Error Description: 
**Collecting lance
  Downloading lance-1.2.1-py3-none-any.whl (22 kB)
Installing collected packages: lance
Successfully installed lance-1.2.1
AttributeError                            Traceback (most recent call last)
[<ipython-input-1-41cc96aab25f>](https://localhost:8080/#) in <cell line: 3>()
      1 get_ipython().system('pip install lance')
      2 import lance
----> 3 dataset = lance.dataset('codeparrot_1M.lance/')
      4 print(dataset.count_rows())
 AttributeError: module 'lance' has no attribute 'dataset'**

(colab)
Code: 
**!pip install lance
import lance
dataset = lance.dataset('codeparrot_1M.lance/')
print(dataset.count_rows())**

**Comment by tanaymeh** [comment]
Hi @Aryan-Deshpande,

Can you try uninstalling lance and installing it like this?

```bash
$ pip install --upgrade pyarrow
$ pip install --upgrade pylance
```

The correct name for Lance for pip install is "pylance".

Please let us know if it works. Cheers!

**Comment by Aryan-Deshpande** [comment]
![image](https://github.com/lancedb/lance/assets/72693780/da1c556a-3e02-43ad-9a68-36f5b3a51f65)

Hey @tanaymeh , The issue still persists

**Comment by changhiskhan** [comment]
you may have to uninstall the pypi package named `lance` first. Then reinstall `pip install pylance` afterwards

**Comment by tanaymeh** [comment]
@Aryan-Deshpande It works fine when I run it in a fresh Google Colab session. 

Can you try deleting the colab runtime and starting afresh? Thanks!

**Comment by Aryan-Deshpande** [comment]
![image](https://github.com/lancedb/lance/assets/72693780/877e265d-b86e-404b-851e-956599dd41dd)
The previous issue was fixed, but I encountered another error

It says dataset was not found at the location content/codeparrot_1M.lance

**Comment by tanaymeh** [comment]
@Aryan-Deshpande It runs fine for me when I download the dataset following the instructions on the Kaggle Datasets page for the Codeparrot 1M dataset with changes for Google Colab since it downloads everything at `/content`. 

You need to make the necessary changes in the path depending on what platform you are using (for example: `/content/` for Google Colab or `/kaggle/working/` for Kaggle, etc)

Here's the bash script that worked fine for me:
```bash
pip install -q kaggle pyarrow pylance
mkdir ~/.kaggle/
mv /content/kaggle.json ~/.kaggle/kaggle.json  # move kaggle.json API key to ~/.kaggle/ from /content
kaggle datasets download -d heyytanay/codeparrot-1m
mkdir /content/codeparrot_1M.lance/
unzip -qq /content/codeparrot-1m.zip -d /content/codeparrot_1M.lance/
rm /content/codeparrot-1m.zip
```

**Reply by Aryan-Deshpande** [reply]
Ah ok, so do we need to download the dataset first from kaggle then unpack and use it ?. Must have been my misunderstanding, I thought it would download it automatically.

**Comment by tanaymeh** [comment]
Converting this to a discussion since it's not an issue with Lance.
